internal/utils/chaincode.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package utils
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+)
+
+func GetChaincodeDefinition(chaincodeId string, stub shim.ChaincodeStubInterface) (*lifecycle.QueryApprovedChaincodeDefinitionResult, error) {
+	function := "QueryApprovedChaincodeDefinition"
+	args := &lifecycle.QueryApprovedChaincodeDefinitionArgs{
+		Name:     chaincodeId,
+		Sequence: 0, // TODO get current sequence number
+	}
+	argsBytes, err := proto.Marshal(args)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := stub.InvokeChaincode("_lifecycle", [][]byte{[]byte(function), argsBytes}, stub.GetChannelID())
+
+	df := &lifecycle.QueryApprovedChaincodeDefinitionResult{}
+	if err := proto.Unmarshal(resp.Payload, df); err != nil {
+		return nil, err
+	}
+	return df, nil
+}
+
+func ExtractMrEnclaveFromChaincodeDefinition(chaincodeId string, stub shim.ChaincodeStubInterface) (string, error) {
```

Comment:
Probably not for this PR, but i'm wondering whether we do not also need the sequence number. Or do we for now make sure ercc does exactly a single read to lscc for a particular chaincode?  If we do multiple calls, we might get strange (and potentially insecure) behaviour when the chaincode changes? admittedly, the at least lifecycle policy orgs have to sign off a change but probably still worth keeping that in mind? Or do we want to explicitly _not_ support changing chaincode definitions for now (in which case we maybe want to make this explicit in MVP issues?)

Reply:
well, this should not happen. When the ercc transaction is processed, it will work on a ledger snapshot. I hope that this snapshot is also shared with `_lifecycle` and thus concurrent changes will not impact the execution.

I am wondering how we are supplying the sequence number?! Is this an input of ercc? Should ercc always use the most recent sequence number?

Reply:
> well, this should not happen. When the ercc transaction is processed, it will work on a ledger snapshot.

My concern was consistency _across_ transactions, e.g., enclave1 has a transaction referencing lscc, chaincode def changes, enclave2 has a transaction references lssc (and doing so without realizing that cc def has changed ...)

> I hope that this snapshot is also shared with `_lifecycle` and thus concurrent changes will not impact the execution.
> 
> I am wondering how we are supplying the sequence number?! Is this an input of ercc? Should ercc always use the most recent sequence number?

as you register a chaincode, ercc could retrieve (current) sequence number from chaincode def and ensure during any ercc update tx  that it is still same sequence number?